src/role_harvester.js

```diff
@@ -20,6 +20,16 @@ roles.harvester.stayInRoom = true;
 roles.harvester.buildRoad = true;
 roles.harvester.boostActions = ['capacity'];
 
+roles.harvester.getPartConfig = function(room) {
+  let datas = {layout: [MOVE, MOVE, WORK, CARRY],
+    maxEnergyUsed: 1500,
```

Comment:
if prototype_room_utils.js had a storageMoreThanThreshold shorhand, your could add
```js
  check: 'utils.storageMoreThanThreshold', // maybe rename to storeAboveThreshold
  onSucces: {addSufix: [WORK, MOVE]},
```
and maybe there will be some where else:
```js
  check: 'utils.shouldDefend',
  onFail: {addSufix: [WORK, MOVE]},
  onSucces: {addPrefix: [TOUGH, TOUGH]},
```
or
```js
  check: 'utils.shouldDefend',
  onFail: {replace: {prefix:[TOUGH, TOUGH], replacement: [WORK, MOVE]}},
```